src/arrayMethodJoin.js

```diff
@@ -4,8 +4,43 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const length = this.length;
+    let firstElement = String(this[0]);
+    let separatorValue = separator;
+
+    if (length === 0) {
+      return '';
+    }
+
+    if (firstElement === 'undefined' || firstElement === 'null') {
+      firstElement = '';
+    }
+
+    if (length === 1) {
+      return firstElement;
+    }
+
+    if (separatorValue === undefined) {
+      separatorValue = ',';
+    }
+
+    if (separatorValue === null) {
```

Comment:
Also no need for this check, because if separator is null and you add it to your joined string, it will become a string
![image](https://github.com/mate-academy/js_array-method-join/assets/46566190/aff23677-5829-42e7-8cd2-f87b989f7c39)

Reply:
got it